### PR TITLE
Update engine without legacy code

### PR DESCRIPTION
**Pull request by irisfaraway**
https://eaflood.atlassian.net/browse/RUBY-1592

There are some components (notably Reform) that have been removed from the engine but are still being refactored out of the back office. So I've added some things back in temporarily.

Also noticed that we'll need to change the way we store the AD status in the engine - this will be done in an upcoming PR.

### CODE REVIEW

File: spec/factories/submitted_enrollments_factory.rb [hunk 1]
```diff
@@ -9,7 +9,14 @@
     next if ot.to_sym == :unknown
 
     factory :"submitted_#{ot}", parent: :confirmed_random_pending do
+      reference_number { FloodRiskEngine::ReferenceNumber.create }
+      submitted_at { Time.current }
+
       after(:create) do |object|
+        object.enrollment_exemptions.each do |ee|
```
Comment (tobyprivett) [inline spec/factories/submitted_enrollments_factory.rb:16]: Just checked this with my WIP branch (feature/enrollment-actions) and it changes the default status of the enrollment_exemption. Should be okay to change my code, just wondering what this change does?

**Review by tobyprivett — APPROVED** [review]
Hi, just a comment about the factory change

File: spec/factories/submitted_enrollments_factory.rb (hunk 1 above)
Comment (tobyprivett) [reply spec/factories/submitted_enrollments_factory.rb:16]: This is the bit: `ee.status = 1`

Comment (irisfaraway) [reply spec/factories/submitted_enrollments_factory.rb:16]: Ahh, it wasn't intended to change the status – more replicate the behaviour of the old `enrollment.submit` function. I'll see if I can tweak it.

Comment (irisfaraway) [reply spec/factories/submitted_enrollments_factory.rb]: Just checked and turns out we didn't need that change 🤷‍♀️ Maybe I changed a test that was relying on it later or something!

**Pull request merged**